### PR TITLE
fix(): add type for useMeilisaerch, closes #21 

### LINKED ISSUE #21: Typescript issue
useMeilisearchClient returns any type

image

### PR DESCRIPTION
**Pull request by NarHakobyan**

### CODE REVIEW

**Comment by NarHakobyan** [comment]
@joewinger @xlanex6 could you please review and publish this? :)

**Comment by xlanex6** [comment]
hi @NarHakobyan 

yes I can approve your PR
thanks for the works 

As you maybe knows, Meilisearch was release in v1 couple days ago, and as Ambassador I knew it for a while. 
the plan was to make a massive update for v1 and include better types support.
I should perhaps tell you earlier.

Why you published https://www.npmjs.com/package/nuxt-meilisearch-typed ? 
It could confuse final users, not ? 
Can we work together instead of works in two different packages ?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "nuxt-meilisearch",
-  "version": "0.1.7",
+  "name": "nuxt-meilisearch-typed",
```
Comment (xlanex6) [inline package.json:2]: The name should be the same 
`nuxt-meilisearch`

**Comment by NarHakobyan** [comment]
Hi @xlanex6 We thought it's not maintainable anymore, I've deleted this module, Also I think it's better to close this and as you said rewrite all.

**Comment by NarHakobyan** [comment]
@xlanex6 also please let's make sure ais-instant-search components will support types, it's too hard to go into the lib and check code each time when using typescript.

**Comment by xlanex6** [comment]
Sure @NarHakobyan !

We can add type info into docs site, by that way, final users could see typescript usage
ais-instant-search depends about Algolia, so we should digging on their repo / lib to find types declaration.

also
you say `we` on your previous comment
Can you explain a bit more, do you use it in production, live app ?

**Comment by NarHakobyan** [comment]
We wanted to use it on production, but because of some issues (returning type of useMeilisearchClient, or ais-instant-search components) we decided to write it from scratch instead of using the lib, also zuseMeilisearchClientz function doesn't return meilisearch instant so that was also a cause of removing the lib.

**Comment by xlanex6** [comment]
So we can create a `typed` branch and works on typed max as we can 

What do you think about it ?

**Comment by NarHakobyan** [comment]
@xlanex6 I know many ways to use, thanks for your advice, it was necessary to create and publish version on NPM. But now it's useless so I deleted it.

**Comment by NarHakobyan** [comment]
does anyone use Algolia lib with Meilisearch? for example we created it from scratch to reduce http requests count by adding some debounce, Alos as I know there is not way to add types on slot `v-slot="{ items }"`, so it's going to be any type right? 🤔

**Pull request closed without merging**